deepfence_server/pkg/integration/splunk/splunk.go

```diff
@@ -23,64 +31,93 @@ func New(ctx context.Context, b []byte) (Splunk, error) {
+	} else if numRoutines > 10 {
+		numRoutines = 10
+	}
+
+	senderChan := make(chan []byte, 500)
+	var wg sync.WaitGroup
+	for i := 0; i < numRoutines; i++ {
+		wg.Add(1)
+		go s.Sender(senderChan, &wg)
+	}
+
+	defer func() {
+		close(senderChan)
+		wg.Wait()
+		log.Info().Msgf("SPLUNK::SendNotification complete, numRoutines:%d", numRoutines)
+	}()
+
 	var buffer bytes.Buffer
 	for _, payload := range msg {
 		currentMicro := time.Now().UnixMicro()
 		payload["timestamp"] = currentMicro
 		payload["@timestamp"] = currentMicro
-		// Create a new HEC event with the data
+
 		event := HECEvent{Event: payload}
-		// Marshal the HEC event to JSON
 		jsonBytes, err := json.Marshal(event)
 		if err != nil {
-			fmt.Println("Failed to marshal HEC event to JSON", err)
+			log.Info().Msgf("Failed to marshal HEC event to JSON, %v", err)
 			continue
 		}
+
 		// Send out bytes in buffer immediately if the limit exceeded after adding this event
 		if buffer.Len()+len(jsonBytes) > MaxContentLength {
-			s.sendRequest(buffer)
+			b := make([]byte, buffer.Len())
+			copy(b, buffer.Bytes())
+			senderChan <- b
 			buffer.Reset()
 		}
 		buffer.Write(jsonBytes)
 	}
+
 	if buffer.Len() > 0 {
-		s.sendRequest(buffer)
+		senderChan <- buffer.Bytes()
 	}
+
 	return nil
 }
 
-// HECEvent represents an event for the Splunk HTTP Event Collector (HEC) API
-type HECEvent struct {
-	Event map[string]interface{} `json:"event"`
-}
+func (s Splunk) Sender(in chan []byte, wg *sync.WaitGroup) {
+	defer wg.Done()
+	authToken := "Splunk " + s.Config.Token
 
-func (s Splunk) sendRequest(buffer bytes.Buffer) error {
-	client := utils.GetInsecureHttpClient()
-	// Create a new request to send the JSON data to Splunk
-	req, err := http.NewRequest("POST", s.Config.EndpointURL, bytes.NewReader(buffer.Bytes()))
-	if err != nil {
-		fmt.Println("Failed to create HTTP request", err)
-	}
-	req.Header.Set("Authorization", "Splunk "+s.Config.Token)
-	// req.Header.Set("Content-Type", "application/json")
+SenderLoop:
+	for {
+		select {
+		case data, ok := <-in:
+			if !ok {
+				break SenderLoop
+			}
 
-	// Send the request to Splunk
-	resp, err := client.Do(req)
-	if err != nil {
-		fmt.Println("Failed to send data to Splunk", err)
-		return err
-	}
-	defer resp.Body.Close()
+			req, err := http.NewRequest("POST", s.Config.EndpointURL, bytes.NewReader(data))
+			if err != nil {
+				log.Info().Msgf("Failed to create HTTP request: %v", err)
+				continue SenderLoop
+			}
+			req.Header.Set("Authorization", authToken)
+			resp, err := s.client.Do(req)
+			if err != nil {
+				log.Info().Msgf("Failed to send data to Splunk: %v", err)
+				continue SenderLoop
+			}
+			defer resp.Body.Close()
 
-	// Check the response status code
-	if resp.StatusCode != http.StatusOK {
-		fmt.Println("Failed to send data to Splunk", resp.Status)
+			// Check the response status code
+			if resp.StatusCode != http.StatusOK {
+				log.Info().Msgf("Failed to send data to Splunk %s", resp.Status)
+			}
+		}
```

Comment:
It will be cleaner (less indentation) to have this code outside the `select`
Since you have only 1 branch, we can simply have:
```
var data any
var ok bool
select {
case data, ok = <-in:
}
...
```

Reply:
Done